dotcom-rendering/src/components/ElectionTrackers/ValuesWithChange.tsx

```diff
@@ -0,0 +1,202 @@
+	 * element, so each value's `name` should be unique relative to the other
+	 * sections. Note that due to the constraints of the design this cannot be
+	 * longer than about 7 characters, otherwise it will overrun its container.
+	 *
+	 * **Examples:** name of a candidate; name of a party.
+	 */
+	name: string;
+	/**
+	 * **Examples:** seats won by a party; votes won by a candidate.
+	 */
+	value: number;
+	/**
+	 * The change in the value; since the last election, for example. Can be
+	 * positive, negative or zero. If zero, it will not be shown.
+	 */
+	change: number;
+	/**
+	 * The colour used to represent the value. It expects a CSS `color` value
+	 * (e.g. a hex string). To ensure dark mode support a {@linkcode palette}
+	 * colour can be used; i.e. this property can be set to the return value of
+	 * the {@linkcode palette} function.
+	 */
+	colour: string;
+};
+
+const rowGap = 10;
+
+export const ValuesWithChange = ({
+	values,
+	valueDescription,
+	changeDescription,
+}: Props) => (
+	<ul
+		css={{
+			display: 'grid',
+			gridTemplateColumns: 'repeat(auto-fill, minmax(84px, 1fr))',
+			rowGap,
+			'--column-gap': '10px',
+			columnGap: 'var(--column-gap)',
+			/**
+			 * Used to hide parts of the borders between grid items, which are
+			 * created using pseudo-elements.
+			 */
+			overflow: 'hidden',
+			listStyle: 'none',
+			marginTop: space[2],
+			padding: 0,
+			[from.mobileLandscape]: {
+				'--column-gap': '20px',
+			},
+		}}
+	>
+		{values.map((value) => (
+			<Value
+				key={value.name}
+				value={value}
+				valueDescription={valueDescription}
+				changeDescription={changeDescription}
+			/>
+		))}
+	</ul>
+);
+
+const Value = ({
+	value,
+	valueDescription,
+	changeDescription,
+}: {
+	value: ValueWithChange;
+	valueDescription: Props['valueDescription'];
+	changeDescription: Props['changeDescription'];
+}) => (
+	<li
+		style={{
+			'--before-background-colour': value.colour,
+		}}
+		css={{
+			whiteSpace: 'nowrap',
+			color: palette('--values-with-change-value'),
+			position: 'relative',
+			['&:before']: {
+				content: '""',
+				display: 'block',
+				height: 4,
+				width: 32,
+				borderRadius: 20,
+				backgroundColor: 'var(--before-background-colour)',
+			},
+			['&:after']: {
+				content: '""',
+				position: 'absolute',
+				top: -rowGap,
+				left: 'calc(-0.5 * (var(--column-gap) + 1px))',
+				backgroundColor: palette('--values-with-change-border'),
+				height: `calc(100% + ${rowGap}px)`,
+				width: 1,
+			},
+		}}
+	>
+		<Name name={value.name} />
+		<ValueText value={value.value} valueDescription={valueDescription} />
+		<Change change={value.change} changeDescription={changeDescription} />
+	</li>
+);
+
+const ValueText = ({
+	value,
+	valueDescription,
+}: {
+	value: ValueWithChange['value'];
+	valueDescription: Props['valueDescription'];
+}) => (
+	<span
+		aria-label={valueDescription}
+		css={{
+			['&']: css(headlineBold24),
+			lineHeight: 1,
+		}}
```

Comment:
Could this have been written like this as well?
```
css={[
  headlineBold24,
  {
    lineHeight: 1,
  },
]}
```

Reply:
Yes, and I have written it like that in the past. I just figured out that it could also be written like this, and I thought it was perhaps a little clearer using the commonly recognised nesting selector (`&`), and having everything in one object rather than an array. It also means less indentation, and I think makes it a little easier to read in situations where you also have media queries (see the `Abbreviation` and `ChangeText` components in #13921).

What do you think?

Reply:
Each of the presets has an object version so you should be able to spread the properties like so:

```js
css={{
  ...headlineBold24Object,
  lineHeight: 1,
}}

Reply:
This is a great suggestion, I didn't realise we had this API, thanks @jamesmockett !